frontend/src/pages/home/courses/StudentCourses.tsx

```diff
@@ -0,0 +1,54 @@
+import { useEffect, useState } from "react";
+import NoCoursesFound from "../../../../src/assets/NoCoursesFound.svg";
+interface Course {
+  courseid: string;
+  coursecode: string;
+  coursename: string;
+  coursedescription: string;
+  coursecredits: number;
+  courseteacher: string;
+}
+
+export default function StudentCourses() {
+  const [courses, setCourses] = useState<Course[]>([]); // Initialize with an empty array
+
+  useEffect(() => {
+    const fetchCourses = async () => {
+      const response = await fetch("/api/courses");
+      const data = await response.json();
```

Comment:
Ideally, maybe add a try catch clause here in case the fetch fails in this scenario

Reply:
Done